lib/virt_autotest/utils.pm

```diff
@@ -137,31 +138,34 @@ sub print_cmd_output_to_file {
 }
 
 sub download_script_and_execute {
-    my (%args) = @_;
+    my ($script_name, %args) = @_;
     $args{output_file} //= "$args{script_name}.log";
     $args{machine} //= 'localhost';
 
-    download_script(script_name => $args{script_name}, script_url => $args{script_url}, machine => $args{machine});
-    my $cmd = "~/$args{script_name}";
+    download_script($script_name, script_url => $args{script_url}, machine => $args{machine});
```

Comment:
Why mixed use of positional and named arguments ?

Reply:
because $script_name is mandatory, others are optional, like 
`script_run($cmd [, timeout => $timeout] [, output => ''] [, quiet => $quiet] [, die_on_timeout => -1]);`